src/main/java/com/github/manolo8/darkbot/utils/login/LoginData.java

```diff
@@ -1,13 +1,32 @@
 package com.github.manolo8.darkbot.utils.login;
 
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import com.github.manolo8.darkbot.backpage.BackpageManager;
+import com.github.manolo8.darkbot.config.Config;
+import com.github.manolo8.darkbot.config.ConfigEntity;
+import com.github.manolo8.darkbot.utils.I18n;
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
 
 public class LoginData {
-    private static final Pattern USER_ID_PATTERN = Pattern.compile("userID=(\\d+)");
+    private static final Type PARAMS_TYPE = new TypeToken<Map<String, String>>() {}.getType();
+    private static final Map<String, BiFunction<Config, String, String>> REPLACEMENTS = new HashMap<>() {{
+        put("display2d", (config, defValue) -> config.BOT_SETTINGS.API_CONFIG.USE_3D ? "1" : "2");
+        put("autoStartEnabled", (config, defValue) -> "1");
+        put("lang", (config, defValue) -> Optional.of(I18n.getLocale().getLanguage())
+                .filter(l -> !l.isEmpty())
+                .filter(l -> config.BOT_SETTINGS.API_CONFIG.FORCE_GAME_LANGUAGE)
+                .orElse(defValue));
+    }};
```

Comment:
i feel like this has no place in login data

login data is supposed to be a simple object for storing the user's login credentials for use in the bot, this is just logic that should be handled elsewhere.

Also i think we should make better use of ConfigSetting, i think we could declare these in a cleaner way